libs/types/src/system_settings.ts

```diff
@@ -15,20 +15,24 @@ import {
   StartingCardLockoutDurationSecondsSchema,
 } from './auth';
 
+interface AuthSettings {
+  readonly arePollWorkerCardPinsEnabled: boolean;
+  readonly inactiveSessionTimeLimitMinutes: InactiveSessionTimeLimitMinutes;
+  readonly numIncorrectPinAttemptsAllowedBeforeCardLockout: NumIncorrectPinAttemptsAllowedBeforeCardLockout;
+  readonly overallSessionTimeLimitHours: OverallSessionTimeLimitHours;
+  readonly startingCardLockoutDurationSeconds: StartingCardLockoutDurationSeconds;
+}
+
 /**
  * Settings for various parts of the system that are not part of the election
  * definition. These settings can be changed without changing the election hash
  * (and therefore not needing to reprint ballots, for example).
  */
 export interface SystemSettings {
-  arePollWorkerCardPinsEnabled: boolean;
-  inactiveSessionTimeLimitMinutes: InactiveSessionTimeLimitMinutes;
-  numIncorrectPinAttemptsAllowedBeforeCardLockout: NumIncorrectPinAttemptsAllowedBeforeCardLockout;
-  overallSessionTimeLimitHours: OverallSessionTimeLimitHours;
-  startingCardLockoutDurationSeconds: StartingCardLockoutDurationSeconds;
+  readonly auth: AuthSettings;
 }
```

Comment:
So there's one more place where we need to account for this updated structure, namely all the `constructAuthMachineState` functions in the app backends:

https://github.com/votingworks/vxsuite/blob/500be04f064cda0511843c578b5c278b35addf08/apps/admin/backend/src/app.ts#L113-L125

TS isn't complaining because all the machine state fields are optional:

https://github.com/votingworks/vxsuite/blob/500be04f064cda0511843c578b5c278b35addf08/libs/auth/src/dipped_smart_card_auth_api.ts#L50-L60

Retroactively realizing the risk that can come with optional fields 😅 Might make sense to just mark the relevant fields as required and then make it the app's responsibility to fall back to the defaults when there are no system settings (i.e. before a machine has been configured)

Reply:
Ooh nice catch!

I think your suggestion to fix it makes sense. The other idea that comes to mind is having a test in each app that ensures auth is correctly integrated.

Reply:
^ That makes sense! I'm fine with either

Reply:
Implemented required fields here: https://github.com/votingworks/vxsuite/pull/3860. It turns out there were tests, but the tests never set the system settings during machine configuration, so they weren't catching this.